### PR TITLE
Get default logging level and log file from default opts dict

### PR DESCRIPTION
**Pull request by vutny**
### What does this PR do?

It establishes a single source of truth for default logging level and log file path setting for Salt components. In other words, the motivation is to get rid of hard-coded values in `salt/utils/parsers.py` which are not taken into account in any case, but this could give some side-effects like described below.
### What issues does this PR fix or reference?

For example, the `salt-call` and `salt-cloud` commands are displaying wrong information about default log-level in their online help.

I have completely empty `/etc/salt/minion` file and nothing in `/etc/salt/minion.d`.
Help says that default log level is `info`:

```
# salt-call --help
Usage: salt-call [options] <function> [arguments]

Salt call is used to execute module functions locally on a minion

...
  Logging Options:
    Logging options which override any settings defined on the
    configuration files.

    -l LOG_LEVEL, --log-level=LOG_LEVEL
                        Console logging log level. One of 'all', 'garbage',
                        'trace', 'debug', 'profile', 'info', 'warning',
                        'error', 'critical', 'quiet'. Default: 'info'.
    --log-file=LOG_FILE
                        Log file path. Default: '/var/log/salt/minion'.
    --log-file-level=LOG_LEVEL_LOGFILE
                        Logfile logging log level. One of 'all', 'garbage',
                        'trace', 'debug', 'profile', 'info', 'warning',
                        'error', 'critical', 'quiet'. Default: 'info'.
```

But actually it is not:

``` yaml
# salt-call config.get log_level
local:
    warning
```
### Tests written?

No

### CODE REVIEW

**Comment by cachedout** [comment]
I think is is all likely fine but I believe it should go in develop instead of in a point release. @vutny what do you think about resubmitting this PR on the develop branch?

**Comment by vutny** [comment]
@cachedout At first look this was a minor issue to me. I've spotted that Salt commands are displaying wrong default log-level settings, instead of real default and documented values.

But when I looked at code in `salt/utils/parsers.py` I've found that there are just hard-coded log level and file default settings for each CLI tool. And they are different for `salt-call` and `salt-cloud` option parsers than the settings defined in `salt/config/__init__.py`.

The problem here is that the only method which honors such settings is `_mixin_setup` in `LogLevelMixIn` class when it constructs help messages for those options. In all other cases the defaults from `salt.config` module are taking precedence. Local defaults are ignored because config dictionaries already have instantiated the settings from constants even if user didn't redefined them.

Currently, only the help messages are affected (this is all I've found so far), but it's very easy to forget/miss the fact that different logging settings are defined twice in the code, which may lead to a situation where some parts of the code would use local defaults, but others use "real" ones.

### TL;DR
I believe this should be fixed in earliest branch possible, but it appears that the `2015.8` is left behind to much. That's why I think this PR should go to the `2016.3` since it will be kept maintained long enough.

**Comment by cachedout** [comment]
@s0undt3ch Any comments on this?

**Comment by isbm** [comment]
@cachedout I think this is a good idea. I went through the PR and seems like it does the right thing. 

Oh, and this can be also done to 2015.8, if @vutny does not mind to back-port that there. :smile:

**Comment by cachedout** [comment]
@isbm Sounds totally reasonable.

@vutny Any chance you could make a unit test here?

**Comment by vutny** [comment]
Thanks for the feedback @isbm !

@cachedout , I'll try to get my hands on writing some unit test(s) for this particular case. Stay tuned :smile:

**Comment by cachedout** [comment]
@vutny Great! We'll also need this rebased, since it's now conflicting. Thanks.

**Comment by cachedout** [comment]
Thanks for the rebase. Let me know when you have tests ready and we'll get this in. Thanks, @vutny

**Review by s0undt3ch — APPROVED** [review]

**Comment by cachedout** [comment]
Hi @vutny 

No hurry here but just checking with you on the test progress. Thanks!

**Comment by vutny** [comment]
@cachedout Thanks for the ping back.

I'm working on the tests and actually discovering many little inconsistencies in the code during the process :smile:
I think I'm close to finish it soon enough.

**Comment by cachedout** [comment]
Thanks, @vutny. Just let me know when you're ready to go.

**Comment by vutny** [comment]
@cachedout Finally, this day has come! :smile: 
I've added 130 unit tests for parsing logging options expressed via CLI and configuration for all Salt executables.

Please review, now it's ready for merge.

**Comment by cachedout** [comment]
Great work here, @vutny. We really appreciate this one!
